IN-CORE/pyincore
# 458 check offline mode and disable method interact with services

Wrote custom decorator to check "self.client.offline" flag. This allows the flexibility to add to only those methods interact with the services.

--------
To test:
You can either try the below one example 
```
from pyincore import IncoreClient, SpaceService

if __name__ == "__main__":
    # client = IncoreClient(offline=True)
    client = IncoreClient()
    spacesvc_offline = SpaceService(client)
    metadata = spacesvc_offline.get_spaces()
```
you should see an error message
```
ValueError: Service is not available in offline mode.
```

For more comprehensive test, run some of the incore-docs notebooks
E.g. in **create_earthquake.ipynb** change "offline=True" in client, and **observe failure**. 
E.g. try **session4-building-dmg-recovery-local-data.ipynb**, should **pass with no issue**
